ui.apps/src/main/content/jcr_root/apps/core/cif/components/commerce/product/v1/product/variantselector.html

```diff
@@ -15,16 +15,18 @@
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/-->
 <template data-sly-template.variants="${ @ product}">
     <sly data-sly-list.variantAttribute="${product.variantAttributes}">
+        <sly data-sly-test.hasSwatchData="${variantAttribute.values[0].swatchType}"></sly>
+        <sly data-sly-test.isColorOption="${hasSwatchData && variantAttribute.values[0].swatchType == 'COLOR'}"></sly>
         <div class="option__root">
             <h3 class="option__title"><span>${variantAttribute.label}</span></h3>
             <div
-                data-sly-test.isColor="${variantAttribute.label == 'Color'}"
+                data-sly-test.isColor="${isColorOption || variantAttribute.label == 'Color'}"
                 data-id="${variantAttribute.id}"
                 class="swatchList__root tileList__root"
             >
                 <sly data-sly-list.variantValue="${variantAttribute.values}">
                     <button
-                        class="swatch__root tile__root clickable__root"
+                        class="${'swatch__root clickable__root tile__root{0}{1}' @ format=[hasSwatchData ? ' tile__root--' : '', hasSwatchData ? variantValue.cssClassModifier : ''], context='text'}"
```

Comment:
The modifier should be applied also for attributes that don't have swatch data but have the label 'Color'.

Reply:
if there is no swatch data then there is no css class modifier in the variant value